src/main/scala/intent/core/expectations/contain.scala

```diff
@@ -76,3 +76,40 @@ class ArrayContainExpectation[T](expect: Expect[Array[T]], expected: T)(
   def evaluate(): Future[ExpectationResult] =
     val actual = expect.evaluate()
     evalToContain(actual, expected, expect, "Array")
+
+class MapContainExpectation[K, V](expect: Expect[Map[K, V]], expected: Seq[Tuple2[K, V]])(
+  given
+    eqq: Eq[V],
+    fmt: Formatter[V]
+) extends Expectation with
+
+  def evaluate(): Future[ExpectationResult] =
+    val actual = expect.evaluate()
+    var failingKeys = Seq[Tuple2[K, V]]()       // When the key failing (missing or present when negated)
+    var failingValues = Seq[Tuple3[K, V, V]]()  // When the key is OK but the value does not match
+
+    expected.foreach { expectedPair =>
+      actual.get(expectedPair._1) match {
+        case None if expect.isNegated =>
+          () // OK = NOOP
+        case None if !expect.isNegated =>
+          failingKeys :+= expectedPair
+        case Some(v) if expect.isNegated =>
+          failingKeys :+= expectedPair
+        case Some(v) if !expect.isNegated   =>
+          if !eqq.areEqual(v, expectedPair._2) then failingValues :+= (expectedPair._1, expectedPair._2, v)
+      }
+    }
+
+    Future.successful:
+      if failingKeys.isEmpty && failingValues.isEmpty then
+        expect.pass
+      else
+        var message = s"Expected ${describeActualWithContext(actual)} to "
+        if (expect.isNegated) message += "not "
+        message += "contain:\n  "
+        message += failingKeys.map(p => s"${p._1} -> ${p._2}").mkString("\n  ")
+        message += failingValues.map(p => s"${p._1} -> ${p._2} but found ${p._1} -> ${p._3}").mkString("\n  ")
+        expect.fail(message)
+
+  private def describeActualWithContext(actual: Map[K, V]): String = "Map(...)"
```

Comment:
Not much context here... ;-)

Reply:
Nope =)
What is a good context for a Map? `Map(of 23 entries)`,`Map[K, V]` or to print a few keys (similar to Seq, and using the same `cutoff`)?